File: app/src/main/java/com/example/spacetraders/data/views/MainGameActivity.java
package com.example.spacetraders.data.views;

import android.content.Intent;
import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.view.View;
import android.widget.Button;
import android.widget.TextView;

import com.example.spacetraders.R;
import com.example.spacetraders.data.entity.Character;
import com.example.spacetraders.data.models.Interactor;

public class MainGameActivity extends AppCompatActivity {
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main_game);

        TextView systemName = findViewById(R.id.solar_system_name);
        Character character = Interactor.getInteractor().getCharacter();
        systemName.setText(character.getCurrentSolarSystem().getName());

        /*
         * Disables actionbar back button
         */
        if (getSupportActionBar() != null) {
            getSupportActionBar().setDisplayHomeAsUpEnabled(false);
            getSupportActionBar().setHomeButtonEnabled(false);
        }

        //Button solarSystem = findViewById(R.id.button_solarSystem);
        TextView solarSystem = findViewById(R.id.solar_system_name);
        Button mercenary = findViewById(R.id.button_mercenary);
        Button market = findViewById(R.id.button_market);
        Button player = findViewById(R.id.button_player);
        Button travel = findViewById(R.id.button_travel);
        Button shipyard = findViewById(R.id.button_shipyard);

        solarSystem.setOnClickListener((View v) -> {
                Intent intent = new Intent(MainGameActivity.this, SolarSystemActivity.class);
                startActivity(intent);
        });

        mercenary.setOnClickListener((View v) -> {
            Intent intent = new Intent(MainGameActivity.this, MercenaryActivity.class);
            startActivity(intent);
        });


        market.setOnClickListener((View v) -> {
            Intent intent = new Intent(MainGameActivity.this, MarketActivity.class);
            startActivity(intent);
        });

        player.setOnClickListener((View v) -> {
            Intent intent = new Intent(MainGameActivity.this, PlayerActivity.class);
            startActivity(intent);
        });


        travel.setOnClickListener((View v) -> {
            Intent intent = new Intent(MainGameActivity.this, TravelActivity.class);
            startActivity(intent);
        });

        shipyard.setOnClickListener((View v) -> {
            Intent intent = new Intent(MainGameActivity.this, ShipYardActivity.class);
            startActivity(intent);
        });

        mercenary.setOnClickListener((View v) -> {
            Intent intent = new Intent(MainGameActivity.this, MercenaryActivity.class);
            startActivity(intent);
        });
    }

    @Override
    protected void onPause() {
        super.onPause();
        Interactor.getInteractor().saveGame(getApplicationContext());
    }
}

File: app/src/main/java/com/example/spacetraders/data/views/MainGameActivity.kt
package com.example.spacetraders.data.views

import android.content.Intent
import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import android.view.View
import android.widget.Button
import android.widget.TextView

import com.example.spacetraders.R
import com.example.spacetraders.data.entity.GameDifficulty
import com.example.spacetraders.data.entity.Resources
import com.example.spacetraders.data.entity.SolarSystem
import com.example.spacetraders.data.entity.Universe
import com.example.spacetraders.data.entity.Character
import com.example.spacetraders.data.models.Interactor

class MainGameActivity : AppCompatActivity() {
    private var solarSystem: Button? = null
    private var planet: Button? = null
    private var market: Button? = null
    private var player: Button? = null
    private var travel: Button? = null
    private var cargo: Button? = null
    private var shipyard: Button? = null
    internal var systemName: TextView? = null
    internal var character: Character? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main_game)
        systemName = findViewById(R.id.solar_system_name)
        character = Interactor.interactor.character
        systemName!!.text = character!!.currentSolarSystem!!.name

        /*
         * Disables actionbar back button
         */
        if (supportActionBar != null) {
            supportActionBar!!.setDisplayHomeAsUpEnabled(false)
            supportActionBar!!.setHomeButtonEnabled(false)
        }

        solarSystem = findViewById(R.id.button_solarSystem)
        planet = findViewById(R.id.button_planet)
        market = findViewById(R.id.button_market)
        player = findViewById(R.id.button_player)
        travel = findViewById(R.id.button_travel)
        cargo = findViewById(R.id.button_cargo)
        shipyard = findViewById(R.id.button_shipyard)


        solarSystem!!.setOnClickListener { v: View ->
            val intent = Intent(this@MainGameActivity, SolarSystemActivity::class.java)
            startActivity(intent)
        }


        market!!.setOnClickListener { v: View ->
            val intent = Intent(this@MainGameActivity, MarketActivity::class.java)
            startActivity(intent)
        }

        player!!.setOnClickListener { v: View ->
            val intent = Intent(this@MainGameActivity, PlayerActivity::class.java)
            startActivity(intent)
        }


        travel!!.setOnClickListener { v: View ->
            val intent = Intent(this@MainGameActivity, TravelActivity::class.java)
            startActivity(intent)
        }

        shipyard!!.setOnClickListener { v: View ->
            val intent = Intent(this@MainGameActivity, RefuelActivity::class.java)
            startActivity(intent)
        }
    }

    override fun onPause() {
        super.onPause()
        Interactor.interactor.saveGame()
    }
}
